src/main/java/com/marcusslover/plus/lib/item/MenuManager.java

```diff
@@ -1,46 +1,95 @@
 package com.marcusslover.plus.lib.item;
 
-import com.marcusslover.plus.lib.item.Menu.ClickAdapter;
-import com.marcusslover.plus.lib.text.Text;
+import com.marcusslover.plus.lib.events.Events;
 import org.bukkit.Bukkit;
-import org.bukkit.event.EventHandler;
-import org.bukkit.event.Listener;
+import org.bukkit.entity.Player;
 import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.inventory.InventoryCloseEvent;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.InventoryView;
 import org.bukkit.plugin.Plugin;
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
 
 import java.util.HashSet;
-import java.util.Iterator;
-import java.util.Optional;
 import java.util.Set;
 
-public final class MenuManager implements Listener {
+public final class MenuManager {
     private final @NotNull Plugin plugin;
     private final @NotNull Set<@NotNull Menu> menus = new HashSet<>();
 
     public MenuManager(@NotNull Plugin plugin) {
         this.plugin = plugin;
-        Bukkit.getPluginManager().registerEvents(this, plugin);
-        Bukkit.getScheduler().runTaskTimer(plugin, () -> {
-            long now = System.currentTimeMillis();
-            Iterator<@NotNull Menu> iterator = this.menus.iterator();
-            while (iterator.hasNext()) {
-                Menu menu = iterator.next();
-                if (!menu.inventory.getViewers().isEmpty()) {
-                    continue;
-                }
-
-                if (menu.lastActivity < 0) {
-                    iterator.remove();
-                } else {
-                    long time = now - menu.lastActivity;
-                    if (time > 1000 * 60 * 5) { // 5 minutes
-                        iterator.remove();
+        Events.listen(InventoryCloseEvent.class).handler(event -> {
```

Comment:
Technically with some tuning you could use the Plus built in registration system 🤔